Source/MQTTnet/Client/MqttClient.cs

```diff
@@ -222,8 +222,11 @@ private void ThrowIfConnected(string message)
 
             try
             {
-                await WaitForTaskAsync(_packetReceiverTask, sender).ConfigureAwait(false);
-                await WaitForTaskAsync(_keepAliveMessageSenderTask, sender).ConfigureAwait(false);
+                if (!(exception is MqttCommunicationClosedGracefullyException))
```

Comment:
I am not 100% shure if this is correct or if we dont want to wait if any MqttCommunicationException is thrown

Reply:
Mhh I am also not 100 % sure. This wait was added by me because otherwise some exceptions will occur because the threads access the adapter. But the adapter is already disposed.